### PR TITLE
Refactor the time regexping and move the timeboxes to a separate widget

### PR DESCRIPTION
**Pull request by voidus**
We use it in a few places now.

### CODE REVIEW

**Comment by voidus** [comment]
Just noticed I don't need this at the moment actually. Oh well.

File: phprojekt/htdocs/phpr/Timehelper.js [hunk 1]
```diff
@@ -112,4 +112,26 @@ define([
 
         return ret;
     };
+
+    exports.timeRegexp = function() {
```
Comment (exi) [inline phprojekt/htdocs/phpr/Timehelper.js]: This name suggests that it returns a regex when in fact, it returns a string.
timeRegexpString would be clearer

Comment (exi) [inline phprojekt/htdocs/phpr/Timehelper.js]: also why is this a function and not just a string attribute?
Edit: nevermind, overlooked the () at the end.
I think the 'standard' for this case is to put the function in ()'s like
(function(arg) { //do stuff })()

Comment (exi) [inline phprojekt/htdocs/phpr/Timehelper.js]: explanation to this here: http://stackoverflow.com/questions/440739/what-do-parentheses-surrounding-a-javascript-object-function-class-declaration-m

Comment (voidus) [inline phprojekt/htdocs/phpr/Timehelper.js]: Both fixed

File: phprojekt/htdocs/phpr/Timehelper.js [hunk 2]
```diff
@@ -112,4 +112,26 @@ define([
...
+
+    exports.timeRegexpString = (function() {
+        var hours = '([01]?\\d|2[0123])',
+            minutes = '([01-5]\\d)',
+            separator = '[:\\. ]?';
+
+        return '(' + hours + separator + minutes + ')';
+    })();
+
+    exports.parseWithTimeRegexp = function(value) {
```
Comment (exi) [inline phprojekt/htdocs/phpr/Timehelper.js]: i would call it 'parseTime'.
It does not matter that it gets parsed with a regex or something different.

Comment (voidus) [inline phprojekt/htdocs/phpr/Timehelper.js]: fixed

**Pull request merged**